infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/devfile/KubernetesComponentToWorkspaceApplier.java

```diff
@@ -212,20 +212,21 @@ private void applyProjectsVolumes(List<PodData> podsData, List<HasMetadata> comp
     }
 
     for (PodData podData : podsData) {
+      final String podVolumeName = PROJECTS_VOLUME_NAME + "-" + podData.getMetadata().getName();
```

Comment:
can't we hit some name length limit with this?

Reply:
of course, if pod name longer than 52...

Reply:
This might be easier to hit than we think -- pods will get an 18-char suffix from the replicaset and deployment. It's probably better to at least strip the name down to a max length.

Reply:
This is not the resulting pod. Its intermediate pod which put into environment, constructed from user component recipe.
Resulting pod is merged later in the `PodMerger`.
Example:
Devfile:

```
components:
  - mountSources: true
    referenceContent: |
      kind: List
      items:
       -
        apiVersion: v1
        kind: Pod
        metadata:
         name: zowe1
        spec:
         containers:
          -
           image: 'ca-docker-brightside-docker.bintray.io/cli-mainframe-basic-stack:2020-05-26T14-11-41'
           name: cli1
    type: kubernetes
    alias: zowe-cli1
  - mountSources: true
    referenceContent: |
      kind: List
      items:
       -
        apiVersion: v1
        kind: Pod
        metadata:
         name: zowe2
        spec:
         containers:
          -
           image: 'ca-docker-brightside-docker.bintray.io/cli-mainframe-basic-stack:2020-05-26T14-11-41'
           name: cli2
    type: kubernetes
    alias: zowe-cli2
```

So volumes will be `projects-zowe1` and `projects-zowe2`